platform/icons-api/src/main/kotlin/org/jetbrains/icons/api/IconManager.kt

```diff
@@ -0,0 +1,9 @@
+// Copyright 2000-2025 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.icons.api
+
+interface IconManager {
+  fun loadIcon(id: IconIdentifier, path: String, aClass: Class<*>): Icon?
+  fun registerIcon(id: IconIdentifier, icon: Icon)
+}
+
+interface IconIdentifier
```

Comment:
One of the things we can solve with new icons API is getting rid of dependency on javax.swing from *.psi modules. It seems that `javax.swing.Icon` is the only class from javax.swing which is used there (`PsiElement` interface extends `Iconable` with `getIcon` method). So if we introduce `IconIdentifier`, we can use it in `PsiElement` instead of `javax.swing.Icon` and eventually get rid of the latter.

However, to do that we probably need to put `IconIdentifier` in a different module, so it can be added to dependencies of *.psi modules without having any actual rendering API there.